app/models/merchant.rb

```diff
@@ -1,4 +1,25 @@
 class Merchant < ApplicationRecord
   has_many :items
-  validates :name, presence: true 
+  has_many :invoice_items, through: :items
+  has_many :invoices, through: :invoice_items
+  has_many :transactions, through: :invoices
+  has_many :customers, through: :invoices
+  validates :name, presence: true
+
+  def top_customers
```

Comment:
Thats awesome you were able to get this!  I know it was super challenging.  I like how you made it an instance variable in Merchant because it refers to top customers for a particular instance of a merchant.